services/libs/opensearch/src/service/opensearch.service.ts

```diff
@@ -0,0 +1,378 @@
+import {
+  IndexVersions,
+  OPENSEARCH_INDEX_MAPPINGS,
+  OPENSEARCH_INDEX_SETTINGS,
+  OpenSearchIndex,
+} from '../types'
+import { IS_DEV_ENV } from '@crowd/common'
+import { Logger, LoggerBase } from '@crowd/logging'
+import { Client } from '@opensearch-project/opensearch'
+import { IIndexRequest, ISearchHit } from './opensearch.data'
+import { OPENSEARCH_CONFIG } from '../conf'
+import { AwsSigv4Signer } from '@opensearch-project/opensearch/aws'
+
+export class OpenSearchService extends LoggerBase {
+  public readonly client: Client
+  private readonly indexVersionMap: Map<OpenSearchIndex, string> = new Map()
+
+  constructor(parentLog: Logger) {
+    super(parentLog)
+
+    const indexNames = Object.values(OpenSearchIndex)
+    indexNames.forEach((name) => {
+      const version = IndexVersions.get(name)
+      this.indexVersionMap.set(name, `${name}_v${version}`)
+    })
+
+    const config = OPENSEARCH_CONFIG()
```

Comment:
also because there is no configs here you should pass this configuration through constructor